staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4137,6 +4150,27 @@ type ServiceSpec struct {
 	// If this is not specified or empty, no topology constraints will be applied.
 	// +optional
 	TopologyKeys []string `json:"topologyKeys,omitempty" protobuf:"bytes,16,opt,name=topologyKeys"`
+
+	// IPFamilyPolicy represents the dualstack-ness requested or required by this
+	// Service. If there is no value provided, then this Service will be considered
+	// SingleStack (single IPFamily). Services can be SingleStack (single IPFamily)
+	// PreferedDualStack (two dualstack IPFamilies on dualstack clusters) or (single
+	// IPFamily on singlestack clusters). Services can also be RequireDualStack where
+	// it will have two dualstack IPFamilies on dualstack configured clusters otherwise
+	// (one a single stack cluster) it will fail. The IPFamilies and ClusterIPs assigned
+	// to this service can be controlled by Service.Spec.IPFamilies and Service.Spec.ClusterIPs
+	// respectively
+	// +optional
+	IPFamilyPolicy *IPFamilyPolicyType `json:"ipFamilyPolicy,omitempty" protobuf:"bytes,17,opt,name=ipFamilyPolicy,casttype=IPFamilyPolicyType"`
+
+	// ClusterIPs identifies all the ClusterIPs assigned to or requested for  this
+	// service. ClusterIPs are assigned or reserved  based on the values of
+	// Service.Spec.IPFamilies. Maximum of two entries (dualstack IPs) are
+	// allowed in ClusterIPs. The IPFamily of each ClusterIP must match
+	// values provided in Service.Spec.IPFamilies
+	// +listType=atomic
+	// +optional
+	ClusterIPs []string `json:"clusterIPs,omitempty" protobuf:"bytes,18,opt,name=clusterIPs"`
```

Comment:
Can we put this next to ClusterIP and also document the relationship between them?